extension/js/common/api/account-server.ts

```diff
@@ -60,44 +46,24 @@ export class AccountServer extends Api {
     recipients: ParsedRecipients,
     progressCb: ProgressCb
   ): Promise<UploadedMessageData> => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      // Recipients are used to later cross-check replies from the web
-      //   The message is not actually sent to them now.
-      //   Message is uploaded and a link is retrieved which is sent through Gmail.
-      return await fes.webPortalMessageUpload(encrypted, replyToken, from, recipients, progressCb);
-    } else {
-      return await FlowCryptComApi.messageUpload(await this.getIdToken(), encrypted, progressCb);
-    }
+    const service = await this.getExternalService();
+    return await service.webPortalMessageUpload(encrypted, replyToken, from, recipients, progressCb);
   };
 
   public messageGatewayUpdate = async (externalId: string, emailGatewayMessageId: string) => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      await fes.messageGatewayUpdate(externalId, emailGatewayMessageId);
-    }
+    const service = await this.getExternalService();
+    return await service.messageGatewayUpdate(externalId, emailGatewayMessageId);
   };
 
   public messageToken = async (): Promise<{ replyToken: string }> => {
-    if (await this.isFesUsed()) {
-      const fes = new EnterpriseServer(this.acctEmail);
-      return await fes.webPortalMessageNewReplyToken();
-    } else {
-      const res = await FlowCryptComApi.messageToken(await this.getIdToken());
-      return { replyToken: res.token };
-    }
-  };
-
-  public isFesUsed = async (): Promise<boolean> => {
-    return await isFesUsed(this.acctEmail);
+    const service = await this.getExternalService();
+    return await service.webPortalMessageNewReplyToken();
   };
 
-  private getIdToken = async (): Promise<string> => {
-    const idToken = await InMemoryStore.get(this.acctEmail, InMemoryStoreKeys.ID_TOKEN);
-    if (!idToken) {
-      // user will not actually see this message, they'll see a generic login prompt
-      throw new BackendAuthErr('Missing id token, please re-authenticate');
+  private getExternalService = async (): Promise<ExternalService> => {
```

Comment:
Maybe it's possible to check `isCustomerUrlFesUsed(this.acctEmail)` on `AccountServer` init and set service url there, so we don't need to run `getExternalService` before each request?

Reply:
I thought about that too. One advantage of current system could possibly be that it's a hair more dynamic - it will pick up on a change that happened after the class was initialized. But that's quite an edge case. I agree with what you say. I propose to do it in a separate PR.

Reply:
Yeah, can be a separate issue. Probably we can handle such edge case by re-initialising `AccountServer` when user's `fesUrl` property is changed.

Created issue https://github.com/FlowCrypt/flowcrypt-browser/issues/4917

Reply:
or we can just ignore the edge case, i think.